tests/pos-custom-args/qualified-types/tensorPostfixLambda.scala

```diff
@@ -0,0 +1,82 @@
+import language.experimental.postfixLambda
+
+
+type Shape = List[Int] with _.forall(_ > 0)
+
+extension (s: Shape)
+  //erased ?
+  def nbrElems = s.fold(1)(_ * _)
+  //erased ?
+  def reduce(axes: Shape) = s.zipWithIndex.filter((_, i) => axes.contains(i)).map(_._1)
+
+trait Tensor[T]:
+  val shape: Shape
+
+  //erased ?
+  def sameShape(t: Tensor[T]): Boolean = shape.corresponds(t.shape)(_ == _)
+
+  // TODO: Fix when https://github.com/lampepfl/dotty/issues/18064 is fixed
+  def add(t: Tensor[T] with t.sameShape(this)): Tensor[T] //with _.sameShape(this)
+
+  // TODO: Fix when https://github.com/lampepfl/dotty/issues/17939 is fixed
+  def mean(axes: Shape): Tensor[T] //with _.shape == shape.reduce(axes)
+  def reshape(newShape: Shape with newShape.nbrElems == shape.nbrElems): Tensor[T] //with _.shape == newShape
+
+class TensorImpl[T](val shape: Shape) extends Tensor[T]:
+  def add(t: Tensor[T] with t.sameShape(this)) = this
+  def mean(axes: Shape) = TensorImpl(shape.reduce(axes))
+  def reshape(newShape: Shape with newShape.nbrElems == shape.nbrElems) = TensorImpl(newShape)
+
+
+inline def compiletimeAssert(inline e: Boolean): Unit =
```

Comment:
It is needed semantically, as we want the predicate to be checked at callsite, not in the method body

Reply:
Note that we may decide to still check the method body regardless of `inline`, at which point this would not work, but it at least should not work without `inline`

Reply:
I see, should I leave it and we see what happens once we have the solver ?

Reply:
Okay, let's leave for now. But we'll probably need another version later. Maybe something `def compiletimeAssert(c: Boolean)(using ev: Unit with c)` where `ev` would be synthesized automatically? (Future future work.)

Reply:
There's simpler:
```
def compiletimeAssert2(e: Boolean, u: (Unit with e) = ()): Unit with e = u
```
Since default parameters are checked at application, this should work